controllers/scenario/scenario_adapter.go

```diff
@@ -0,0 +1,146 @@
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package scenario
+
+import (
+	"context"
+	"reflect"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+
+	"github.com/go-logr/logr"
+	applicationapiv1alpha1 "github.com/redhat-appstudio/application-api/api/v1alpha1"
+	"github.com/redhat-appstudio/integration-service/api/v1alpha1"
+	"github.com/redhat-appstudio/integration-service/gitops"
+	"github.com/redhat-appstudio/operator-goodies/reconciler"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// Adapter holds the objects needed to reconcile a Release.
+type Adapter struct {
+	application *applicationapiv1alpha1.Application
+	scenario    *v1alpha1.IntegrationTestScenario
+	logger      logr.Logger
+	client      client.Client
+	context     context.Context
+}
+
+// NewAdapter creates and returns an Adapter instance.
+func NewAdapter(application *applicationapiv1alpha1.Application, scenario *v1alpha1.IntegrationTestScenario, logger logr.Logger, client client.Client,
+	context context.Context) *Adapter {
+	return &Adapter{
+		application: application,
+		scenario:    scenario,
+		logger:      logger,
+		client:      client,
+		context:     context,
+	}
+}
+
+// EnsureCreatedScenarioIsValid is an operation that ensures created IntegrationTestScenario is valid
+// in case it is, set its owner reference
+func (a *Adapter) EnsureCreatedScenarioIsValid() (reconciler.OperationResult, error) {
+	if meta.IsStatusConditionTrue(a.scenario.Status.Conditions, gitops.HACBSIntegrationStatusCondition) {
+		a.logger.Info("IntegrationTestScenario is Invalid.",
+			"Scenario", a.scenario)
+		return reconciler.ContinueProcessing()
+	} else {
+		patch := client.MergeFrom(a.scenario.DeepCopy())
+		err := ctrl.SetControllerReference(a.application, a.scenario, a.client.Scheme())
+		if err != nil {
+			a.logger.Error(err, "Error setting owner reference.",
+				"Application.Name", a.application.Name,
+				"Application.Namespace", a.application.Namespace)
+		}
+		err = a.client.Patch(a.context, a.scenario, patch)
+		if err != nil {
+			a.logger.Error(err, "Failed to update Global Candidate for the Scenario",
+				"Scenario.Name", a.scenario.Name)
+			return reconciler.RequeueWithError(err)
+		}
+		return reconciler.ContinueProcessing()
+	}
+
+}
+
+// EnsureGlobalScenarioUpdated is an operation that ensure the IntegrationTestScenario in the Global Candidate List
+// being updated when the IntegrationTestScenario(ITS) gets detected, set owner reference only for valid ITS
+func (a *Adapter) EnsureGlobalScenarioUpdated() (reconciler.OperationResult, error) {
```

Comment:
I'm not sure what this function is supposed to do, the IntegrationTestScenarios are not supposed to have multiple owner references, or to have global candidates.

Reply:
Yep this has no role/purpose in this context, removing this one.